Постановка целей

MamaevVladimir opened the pull request:
feature https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
Реализована возможность добавлять цели, чтобы формировать бэклог задач для выполнения. 
Для этого добавлены возможности: создать новую цель, обновить цель, удалить цель, 
получить все подзадачи цели по фильтру и получить список целей с фильтрами. 
P.S. Для работы с фильтрами создан интерфейс GoalFilter и классы фильтров, 
которые имплементируют интерфейфейс. 
Условия, которые задаются фильтру, содержатся в записи SearchGoalDto. 
Сделаны некоторые решения, которые отличаются от требований в Jira: 
1) Задание условий обязательно для отбора по фильтрам, поэтому методы контроллеров, работающих с фильтрами: findSubtasksByGoalId(long goalId) и getGoalsByUser(Long userId, GoalFilterDto filter), сделаны с дополнительным параметром SearchGoalDto. 
2) В GoalDto добавлено больше полей, чем в условиях задания, но это не влияет на правильность логики задачи. 
3) Добавлены дополнительные мапперы и дто, которые для данной задачи не обязательные, но могут быть востребованы в других задачах спринта

MamaevVladimir committed:
Постановка целей.
Реализована возможность добавлять цели, чтобы формировать бэклог задач для выполнения.
Для этого добавлены возможности: создать новую цель, обновить цель, удалить цель, получить все подзадачи цели по фильтру и получить список целей с фильтрами.
P.S. Для работы с фильтрами создан интерфейс GoalFilter и классы фильтров, которые имплементируют интерфейфейс. Условия, которые задаются фильтру, содержатся в записи SearchGoalDto.
Сделаны некоторые решения, которые отличаются от требований в Jira:
1) Задание условий обязательно для отбора по фильтрам, поэтому методы контроллеров, работающих с фильтрами: findSubtasksByGoalId(long goalId) и getGoalsByUser(Long userId, GoalFilterDto filter), сделаны с дополнительным параметром SearchGoalDto. 2) В GoalDto добавлено больше полей, чем в условиях задания, но это не влияет на правильность логики задачи.
3) Добавлены дополнительные мапперы и дто, которые для данной задачи не обязательные, но могут быть востребованы в других задачах спринта

MamaevVladimir committed:
Постановка целей.
Реализована возможность добавлять цели, чтобы формировать бэклог задач для выполнения.
Для этого добавлены возможности: создать новую цель, обновить цель, удалить цель, получить все подзадачи цели по фильтру и получить список целей с фильтрами.
P.S. Для работы с фильтрами создан интерфейс GoalFilter и классы фильтров, которые имплементируют интерфейфейс. Условия, которые задаются фильтру, содержатся в записи SearchGoalDto.
Сделаны некоторые решения, которые отличаются от требований в Jira:
1) Задание условий обязательно для отбора по фильтрам, поэтому методы контроллеров, работающих с фильтрами: findSubtasksByGoalId(long goalId) и getGoalsByUser(Long userId, GoalFilterDto filter), сделаны с дополнительным параметром SearchGoalDto. 2) В GoalDto добавлено больше полей, чем в условиях задания, но это не влияет на правильность логики задачи.
3) Добавлены дополнительные мапперы и дто, которые для данной задачи не обязательные, но могут быть востребованы в других задачах спринта

# Conflicts:
#	src/main/java/school/faang/user_service/mapper/SkillMapper.java

CAB4EHKO committed:
BJS2-65023 is done.

CAB4EHKO committed:
BJS2-65023 is done.

CAB4EHKO committed:
BJS2-65023 is done.

CAB4EHKO commented on src/main/java/school/faang/user_service/service/skill/SkillService.java:
```diff
@@ -0,0 +1,30 @@
+
+@Service
+@RequiredArgsConstructor
+public class SkillService {
```
Сделать интерфейс который этот класс бы реализовывал. Принцип инверсии зависимостей.

MamaevVladimir replied:
Интерфейс нужен для определения общего поведения у классов, то есть тех методов,
которые будут реализованы в классах, названия у таких методов одинаковые.
В приложении несколько сервисов, часть методов в них будут одинаковые по сути (создать, обновить, удалить),
но разные по названиям. В проектах на прошлой работе такие сервисы не делали реализациями интерфейса.
Давай эти классы оставим не зависимыми от интерфейсов

Soddik replied:
D из SOLID

CAB4EHKO commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,141 @@
+    public List<GoalDto> findSubtasksByGoalId(long goalId, SearchGoalDto searchGoalDto) {
+        Stream<Goal> subTasks = Optional.ofNullable(goalRepository.findByParent(goalId))
+                .orElseThrow(() -> new IllegalArgumentException("Subtasks not found."));
+        for (GoalFilter goalFilter : goalFilters) {
```
Делаешь фильтрацию, но результаты ни куда не сохраняешь и по факту фильтрация не работает.

MamaevVladimir replied:
Исправил

CAB4EHKO commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,141 @@
+    public GoalDto updateGoal(Long goalId, GoalDto goalDto) {
+        validateId(goalId);
+        if (goalDto.getTitle().isBlank()) {
+            log.error("Title can not be null or empty.", new IllegalArgumentException());
```
Мне кажется ошибку правильнее будет выбрасывать чем просто логировать.

MamaevVladimir replied:
Сделал логирование отдельно, бросание исключения отдельно в виде: 
log.error("Title can not be null or empty.");
throw new IllegalArgumentException();
При проверке заданий с предыдущих модулей менторы одобряли логирование кода,  
и также исключение указывать необходимо. 
Давай в таком варианте напишем, если что-то не совсем правильно, Алексей подскажет?

CAB4EHKO commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,141 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class GoalService {
```
Сделать интерфейс который этот класс бы реализовывал. Принцип инверсии зависимостей.

MamaevVladimir replied:
Интерфейс нужен для определения общего поведения у классов, то есть тех методов,
которые будут реализованы в классах, названия у таких методов одинаковые.
В приложении несколько сервисов, часть методов в них будут одинаковые по сути (создать, обновить, удалить),
но разные по названиям. В проектах на прошлой работе такие сервисы не делали реализациями интерфейса.
Давай эти классы оставим не зависимыми от интерфейсов

Soddik replied:
D из SOLID

CAB4EHKO commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+
+    @GetMapping("/find-goals-by-user/{userId}")
+    public @ResponseBody ResponseEntity<List<GoalDto>>
+    getGoalsByUser(@PathVariable long userId, @RequestBody @NonNull SearchGoalDto searchGoalDto) {
```
Здесь тоже самое с телом запроса в GET-методе

MamaevVladimir replied:
Исправил

CAB4EHKO commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+
+    @GetMapping("/find-subtasks/{goalId}")
+    public @ResponseBody ResponseEntity<List<GoalDto>>
+    findSubtasksByGoalId(@PathVariable long goalId, @RequestBody @NonNull SearchGoalDto searchGoalDto) {
```
GET-методы не должны содержать тело запроса.

MamaevVladimir replied:
Исправил

MamaevVladimir replied:
@RequestBody удалена

MamaevVladimir committed:
bug https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
fixed bugs after cross review

antikler committed:
BJS2-65039 Become Mentor.
release all method of task:
1) getMentees(long userId)
2) getMentors(long userId)
3) deleteMentee(long menteeId, long mentorId)
4) deleteMentor(long menteeId, long mentorId)

CAB4EHKO committed:
BJS2-65023 refactoring.

Commit:
BJS2-65039 Become Mentor: added RequestMapping

Commit:
BJS2-65039 Become Mentor: new realisation task

Commit:
BJS2-65039 Become Mentor: bug fixes

CAB4EHKO committed:
BJS2-65023 refactoring v2.

Commit:
BJS2-65039 Become Mentor: bug fixes after Alexey's code-review

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+    private final GoalService goalService;
+
+    @PostMapping("/create-goal")
+    public @ResponseBody ResponseEntity<GoalDto> createGoal(@RequestParam Long userId,
```
Зачем ResponseBody ?

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+
+import java.util.List;
+
+@Controller
```
Зачем Controller?

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+    public @ResponseBody ResponseEntity<GoalDto> createGoal(@RequestParam Long userId,
+                                                            @RequestBody @NonNull Goal goal) {
+        GoalDto createdGoal = goalService.createGoal(userId, goal);
+        return new ResponseEntity<>(createdGoal, HttpStatus.CREATED);
```
лаконичнее и читабельнее ResponseEntity.status().body()

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+    @DeleteMapping("/delete-goal")
+    public @ResponseBody ResponseEntity<GoalDto> deleteGoal(@RequestParam long goalId) {
+        GoalDto deletedGoal = goalService.deleteGoal(goalId);
+        return new ResponseEntity<>(deletedGoal, HttpStatus.OK);
```
лаконичнее и читабельнее ResponseEntity.noContent().build();
обычно delete не возвращает ничего, т.е. ResponseEntity<Void>

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+    public @ResponseBody ResponseEntity<List<GoalDto>>
+    findSubtasksByGoalId(@PathVariable long goalId, @NonNull SearchGoalDto searchGoalDto) {
+        List<GoalDto> subtask = goalService.findSubtasksByGoalId(goalId, searchGoalDto);
+        return new ResponseEntity<>(subtask, HttpStatus.OK);
```
лаконичнее и читабельнее ResponseEntity.ok(dto)

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+public class GoalController {
+    private final GoalService goalService;
+
+    @PostMapping("/create-goal")
```
/goal/create-goal - излишне
достаточно POST запрос на /goal

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+        return new ResponseEntity<>(createdGoal, HttpStatus.CREATED);
+    }
+
+    @PutMapping("/update-goal")
```
PUT запрос сам говорит что это update, маппинг должен выглядеть так:
```java
@PutMapping("/{goalId}")
public ResponseEntity<GoalDto> updateGoal(@PathVariable("goalId") Long goalId,
                                                            @RequestBody @NonNull GoalDto goal) {
        GoalDto updatedGoal = goalService.updateGoal(goalId, goal);
        return ResponseEntity.ok(updatedGoal);
    }
```

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @DeleteMapping("/delete-goal")
+    public @ResponseBody ResponseEntity<GoalDto> deleteGoal(@RequestParam long goalId) {
```
DELETE запрос сам говорит что это update, маппинг должен выглядеть так:
```java
@DeleteMapping("/{goalId}")
public ResponseEntity<GoalDto> updateGoal(@PathVariable("goalId") Long goalId) {
        goalService.deleteGoal(goalId);
        return ResponseEntity.noContent().build();
    }
```

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+
+    @PostMapping("/create-goal")
+    public @ResponseBody ResponseEntity<GoalDto> createGoal(@RequestParam Long userId,
+                                                            @RequestBody @NonNull Goal goal) {
```
Понимаю что в задаче указана такая сигнатура метода, но так делать не стоит, почему?

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+        goalValidator.validateSkillsToAchieve(goal);
+        if (goalRepository.countActiveGoalsPerUser(userId) == ACTIVE_GOALS_COUNT) {
+            log.error("User can has only {} active goals.", ACTIVE_GOALS_COUNT);
+            throw new IllegalStateException();
```
нужен текст исключения

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+            throw new IllegalStateException();
+        }
+        checkSkillsExistByUserId(userId, goal);
+        Long parentId = goal.getParent().getId() != null ? goal.getParent().getId() : 0L;
```
что значит 0L? может если parentId  = null, то goal является родительским?

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+        }
+        checkSkillsExistByUserId(userId, goal);
+        Long parentId = goal.getParent().getId() != null ? goal.getParent().getId() : 0L;
+        goalRepository.create(goal.getTitle(), goal.getDescription(), parentId);
```
лучше так, 100% будет ID
```java
Goal createdGoal = goalRepository.create(goal.getTitle(), goal.getDescription(), parentId);
saveSkills(userId, goal);
return goalMapper.toDto(goal);
```

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class GoalService {
```
Расставь Transactional там где нужно

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+                .orElseThrow(() -> new IllegalArgumentException("Goal not found."));
+        if (goalById.getStatus().equals(GoalStatus.COMPLETED)) {
+            log.error("The goal status should not be completed during the update.");
+            throw new RuntimeException();
```
Лучше сделать кастомное исключение и конечно добавить текст в него

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+            List<User> usersByGoalId = goalRepository.findUsersByGoalId(goalId);
+            usersByGoalId.forEach(user -> user.getSkills().addAll(new HashSet<>(skillsToAchieve)));
+        }
+        goalById.setTitle(goal.getTitle());
```
обновляются только 2 поля из 4, и нужно использовать маппер

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+        goalById.setStatus(goalDto.getStatus());
+        updateSkills(goalById, goalDto);
+        goalById.setUpdatedAt(LocalDateTime.now());
+        return goalMapper.toDto(goalById);
```
goalById не сохраняется

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+    public GoalDto deleteGoal(long goalId) {
+        Goal goalById = goalRepository.findById(goalId)
+                .orElseThrow(() -> {
+                    log.error("Goal not found.");
```
расширь сообщение указав id и прокинь сообщение в исключение

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+    public List<GoalDto> findSubtasksByGoalId(long goalId, SearchGoalDto searchGoalDto) {
+        Stream<Goal> subTasks = Optional.ofNullable(goalRepository.findByParent(goalId))
+                .orElseThrow(() -> {
+                    log.error("Subtasks not found.");
```
расширь сообщение указав id и прокинь сообщение в исключение

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+                    return new IllegalArgumentException();
+                });
+        goalRepository.deleteById(goalId);
+        return goalMapper.toDto(goalById);
```
зачем возвращать данные которые были удалены?

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    public GoalDto deleteGoal(long goalId) {
+        Goal goalById = goalRepository.findById(goalId)
```
можно использовать existsById

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+        return goalMapper.toDto(goalById);
+    }
+
+    public List<GoalDto> findSubtasksByGoalId(long goalId, SearchGoalDto searchGoalDto) {
```
нэйминг дто

Soddik commented on src/main/java/school/faang/user_service/service/goal/GoalService.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    public List<GoalDto> findSubtasksByGoalId(long goalId, SearchGoalDto searchGoalDto) {
+        Stream<Goal> subTasks = Optional.ofNullable(goalRepository.findByParent(goalId))
```
Зачем обертка в Optional.ofNullable ?
в задаче не прописано, но логично предположить, что для указанного goal может не быть дочерних записей и мы можем вернуть пустой список, т.к. это не является ошибкой

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -0,0 +1,63 @@
+package school.faang.user_service.controller;
```
не тот пакет

MamaevVladimir replied:
Не совсем понимаю. Все пакеты (controller, dto и другие)  сделал 
внутри пакета school.faang.user_service.  Класс контроллера внутри пакета controller

Soddik replied:
В пакете controller.goal cоздайте класс GoalController

Soddik commented on src/main/java/school/faang/user_service/validator/goalvalidator/GoalValidator.java:
```diff
@@ -0,0 +1,35 @@
+    public void validateSkillsToAchieve(Goal goal) {
+        if (Objects.isNull(goal.getSkillsToAchieve()) || goal.getSkillsToAchieve().isEmpty()) {
+            log.error("Can not create goal without skills.");
+            throw new IllegalArgumentException();
```
текстовка

Soddik commented on src/main/java/school/faang/user_service/validator/goalvalidator/GoalValidator.java:
```diff
@@ -0,0 +1,35 @@
+    public void validateSkillsExistInBase(List<Skill> goalSkills, List<Skill> existingSkills) {
+        if (!new HashSet<>(existingSkills).containsAll(goalSkills)) {
+            log.error("Goal skills must be already exist.");
+            throw new IllegalStateException();
```
текстовка

Soddik commented on src/main/java/school/faang/user_service/mapper/GoalMapper.java:
```diff
@@ -0,0 +1,36 @@
+public interface GoalMapper {
+    Goal toEntity(GoalDto goalDto);
+
+    @Mapping(target = "parentId", source = "parent", qualifiedByName = "parentToParentId")
```
можно было через "." маппить:
```java
@Mapping(target = "parentId", source = "goal.parent.id")
```

MamaevVladimir replied:
Интерфейс GoalMapper заменил на абстрактный класс. 
source через "." не маппится, связано это с заменой или нет, не знаю.
Например,  если фиксировать source = "goal.parent.id",
тогда не находит метод из qualifiedByName:
error: Qualifier error. No method found annotated with @Named#value: [ mapParentToParentId ].

Soddik commented on src/main/java/school/faang/user_service/mapper/SkillMapper.java:
```diff
@@ -0,0 +1,30 @@
+    SkillDto toDto(Skill skill);
+
+    @Named("usersToUserIds")
+    default List<Long> usersToUserIds(List<User> users) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/mapper/SkillMapper.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Named("goalsToGoalIds")
+    default List<Long> goalsToGoalIds(List<Goal> goals) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/mapper/UserMapper.java:
```diff
@@ -0,0 +1,30 @@
+    UserDto toDto(User user);
+
+    @Named("goalsToGoalIds")
+    default List<Long> goalsToGoalIds(List<Goal> goals) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/mapper/UserMapper.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Named("skillsToSkillIds")
+    default List<Long> skillsToSkillIds(List<Skill> skills) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/mapper/GoalMapper.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Named("usersToUserIds")
+    default List<Long> usersToUserIds(List<User> users) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/mapper/GoalMapper.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Named("skillsToSkillIds")
+    default List<Long> skillsToSkillIds(List<Skill> skillsToAchieve) {
```
есть дубликаты в других мапперах

Soddik commented on src/main/java/school/faang/user_service/filter/goal/GoalUsersFilter.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Override
+    public Stream<Goal> apply(Stream<Goal> goals, SearchGoalDto searchGoalDto) {
```
что именно должен проверять этот фильтр?

MamaevVladimir replied:
Фильтры по goalUsers, goalSkills удалил, они наверное не нужны здесь.
В слое сервиса через findSubtasksByGoalId() получение подцелей для заданной цели
 и через getGoalsByUser() получение целей заданного пользователя.
Дальше в обоих методах применяются одни и те же фильтры: 
в условиях задачи примерные критерии, по которым проверяет фильтр, 
и эти критерии применяются для обоих методов, поэтому фильтры одни и те же

Soddik commented on src/main/java/school/faang/user_service/filter/goal/GoalSkillsFilter.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Override
+    public Stream<Goal> apply(Stream<Goal> goals, SearchGoalDto searchGoalDto) {
```
что именно делает этот фильтр?

CAB4EHKO committed:
BJS2-65023 refactoring v3.

antikler committed:
BJS2-65039 Become Mentor. Refactor code:
 1) ResponseStatus vs ResponseEntity in MentorshipController.
 2) Transfer annotation @Transactional from private method to public method in MentorshipService.

MamaevVladimir committed:
bug
https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
fixed bugs after mentor`s review

MamaevVladimir committed:
bug
https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
fixed bugs after mentor`s review

MamaevVladimir committed:
bug addition
https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
fixed bugs after mentor`s review part 2

Soddik commented on src/main/java/school/faang/user_service/controller/GoalController.java:
```diff
@@ -13,51 +12,50 @@
 @RequiredArgsConstructor
 public class GoalController {
-    private final GoalService goalService;
+    private final GoalServiceImpl goalService;
```
почему не интерфейс?

Soddik commented on src/main/java/school/faang/user_service/mapper/GoalMapper.java:
```diff
@@ -0,0 +1,64 @@
+
+@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
+@Slf4j
+public abstract class GoalMapper {
```
лучше по классике делать interface и использовать DecorateWith и инжектить репозитории уже в декораторе
ранее я намекал что ты можешь узнать у коллег реализовывал ли кто мапперы для нужных тебе сущностей и если их пр замержены подтянуть изменения себе в ветку через ребейз

MamaevVladimir replied:
Ветку ребэйзил, никто маперы не реализовал. Сделал самостоятельно.
Но те поля, которые не совпадают по названию в target и source, 
в классах Impl перестали устанавливаться.
Сначала убрал @Mapping() аннотации в интерфейсе. 
Эти аннотации переместил в декоратор. Запустил приложение, 
поля не совпадающие по названию не отобразились в Impl.
Затем убрал @Mapping в декораторе, вместо них установка полей через set методы.
В классах Impl эти поля всё так же не отобразились.
Подскажи пожалуйста, что там в коде не добавлено?

Soddik commented on src/main/java/school/faang/user_service/service/skill/SkillService.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.util.List;
+
+public interface SkillService {
```
может уже реализовал кто-то?

Soddik commented on src/main/java/school/faang/user_service/service/skill/SkillServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+
+@Service
+@RequiredArgsConstructor
+public class SkillServiceImpl implements SkillService {
```
может уже реализовал кто-то?

Soddik commented on src/main/java/school/faang/user_service/service/user/UserService.java:
```diff
@@ -0,0 +1,9 @@
+
+import school.faang.user_service.entity.User;
+
+public interface UserService {
```
может уже реализовал кто-то?

MamaevVladimir replied:
Ещё нет

Soddik commented on src/main/java/school/faang/user_service/service/user/UserServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserServiceImpl implements UserService {
```
может уже реализовал кто-то?

MamaevVladimir replied:
Ещё нет

MamaevVladimir committed:
bug
https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
added decorator for mapper, fixed bugs

Soddik reviewed (changes requested):
Перед слиянием нужно разрешить конфликты в коде. Пожалуйста, обнови ветку и устрани их.

MamaevVladimir committed:
bug
https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/353/backlog?selectedIssue=BJS2-65071
added decorator for mapper, fixed bugs

MamaevVladimir committed:
Merge remote-tracking branch 'origin/feature-voldemarmv-BJS2-65071' into feature-voldemarmv-BJS2-65071

MamaevVladimir committed:
merge

MamaevVladimir committed:
fixed bugs with mapper

MamaevVladimir committed:
deleted test class
